src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2377,6 +2418,45 @@ ManageWorkerPool(WorkerPool *workerPool)
 }
 
 
+/*
+ * ConnectionThrottlingRequired contains the logic to decide whether a new connection
+ * can be considered as optional or not. When the function return true, the connection
+ * can be established with optinal flag, else it should not be an optional connection.
+ */
+static bool
+CanUseOptionalConnection(WorkerPool *workerPool)
+{
+	if (list_length(workerPool->sessionList) == 0)
+	{
+		/*
+		 * We definitely need at least 1 connection to finish the execution.
+		 * All single shard queries hit here with the default settings.
+		 */
+		return false;
+	}
+	else if (list_length(workerPool->sessionList) < MaxCachedConnectionsPerWorker)
+	{
+		/*
+		 * Until this session caches MaxCachedConnectionsPerWorker connections,
+		 * this might lead some optional connections to be considered as non-optional
+		 * when MaxCachedConnectionsPerWorker > 1.
+		 *
+		 * However, once the session caches MaxCachedConnectionsPerWorker (which is
+		 * the second transaction executed in the session), Citus would utilize the
+		 * cached connections as much as possible.
+		 */
+		return false;
+	}
+	else if (UseConnectionPerPlacement())
+	{
+		/* user wants one connection per placement, so no throttling is desired */
```

Comment:
should this comment be updated?

Reply:
seems accurate to me, but could use an explanation of why no throttling is desired.